lib/main_common.pm

```diff
@@ -1545,41 +1546,53 @@ sub load_extra_tests {
     loadtest "console/consoletest_setup";
     loadtest 'console/integration_services' if is_hyperv;
     loadtest "console/hostname";
-    if (any_desktop_is_applicable()) {
+    # Extra tests are too long, split the test into subtest according to the
+    # EXTRATEST variable; to maintain compatibility, run all tests if the
+    # variable is equal 1
+    if (check_var('EXTRATEST', 1) && any_desktop_is_applicable()) {
         load_extra_tests_desktop;
     }
+    elsif (check_var('EXTRATEST', 1)) {
+        load_extra_tests_zypper;
+        load_extra_tests_console;
+        load_extra_tests_opensuse;
+        # schedule the docker tests later as it needs the containers module on
+        # SLE>=15 and therefore would potentially pollute other test modules.
+        # Currently for our SLE12 validation tests we are not using a
+        # registered SLE installation so we should not schedule the test
+        # modules.
+        load_extra_tests_docker;
+        load_extra_tests_kdump;
+    }
     else {
-        # Extra tests are too long, split the test into subtest according to the
-        # EXTRATEST variable; to maintain compatibility, run all tests if the
-        # variable is equal 1
-        if (check_var('EXTRATEST', 1)) {
-            load_extra_tests_zypper;
-            load_extra_tests_console;
-            load_extra_tests_opensuse;
-            # schedule the docker tests later as it needs the containers module on
-            # SLE>=15 and therefore would potentially pollute other test modules.
-            # Currently for our SLE12 validation tests we are not using a
-            # registered SLE installation so we should not schedule the test
-            # modules.
-            load_extra_tests_docker;
-            load_extra_tests_kdump;
-        }
-        else {
-            loadtest "console/zypper_ref" unless get_var('EXTRATEST') =~ /zypper/;
-            foreach my $test_name (split(/,/, get_var('EXTRATEST'))) {
-                if (my $test_to_run = main_common->can("load_extra_tests_$test_name")) {
-                    $test_to_run->();
-                }
-                else {
-                    diag "unknown scenario for EXTRATEST value $test_name";
-                }
+        loadtest "console/zypper_ref" unless get_var('EXTRATEST') =~ /zypper/;
```

Comment:
The same as on line 1555, the code can now run even if any_desktop_is_applicable() returns true and that is not desired.